### PR TITLE
[CARBONDATA-3032] Remove carbon.blocklet.size from properties template

### PR DESCRIPTION
**Pull request by xubo245**
1.Remove carbon.blocklet.size from properties template, because carbon.blocklet.size is in V2, V3 use carbon.blockletgroup.size.in.mb or page size
2.change numberComressor to numberCompressor
3.optimize the annotation of method

 - [x] Any interfaces changed?
 No
 - [x] Any backward compatibility impacted?
 No
 - [x] Document update required?
No
 - [x] Testing done
      No
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
No

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/942/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1148/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9206/

**Comment by xuchuanyin** [comment]
For 'carbon.blocklet.size', why not remove this property in code now?

File: docs/sdk-guide.md [hunk 1]
```diff
@@ -24,7 +24,8 @@ CarbonData provides SDK to facilitate
 
 # SDK Writer
 
-In the carbon jars package, there exist a carbondata-store-sdk-x.x.x-SNAPSHOT.jar, including SDK writer and reader.
+In the carbon jars package, there exist a carbondata-store-sdk-x.x.x-SNAPSHOT.jar, including SDK writer and reader. 
+If you want to use SDK, it needs other carbon jar or you can use carbondata-sdk.jar.
```
[the remark this answers is not in the document]
Comment (xubo245) [reply docs/sdk-guide.md]: user use carbondata-store-sdk-x.x.x-SNAPSHOT.jar is not enough, it needs other carbon jar.

but if user use carbondata-sdk.jar, it's enough

**Comment by xubo245** [comment]
@xuchuanyin not sure whether it will has new issue after remove 'carbon.blocklet.size'. and there are some test case to test carbon.data.file.version is V2 scenario

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1147/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9412/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1360/

**Comment by xubo245** [comment]
@xuchuanyin @jackylk Please check this PR again

File: docs/sdk-guide.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (xubo245) [reply docs/sdk-guide.md]: ok, I updated the doc, add all other jar

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1211/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1212/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9476/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1427/

**Comment by xubo245** [comment]
@jackylk I updated.please check

**Pull request closed without merging**